src/coreclr/jit/optimizer.cpp

```diff
@@ -7334,7 +7334,7 @@ bool Compiler::optComputeLoopSideEffectsOfBlock(BasicBlock* blk)
                     if (rhsVN != ValueNumStore::NoVN)
                     {
                         rhsVN = vnStore->VNNormalValue(rhsVN);
-                        if (lvaInSsa(lhsLcl->GetLclNum()))
+                        if (lvaInSsa(lhsLcl->GetLclNum()) && lhsLcl->HasSsaName())
```

Comment:
Yes, that's exactly what I did. The most problematic cases were when we were calling `GetPerSsaData` without this guard, since that will index into bad memory.